gspread/spreadsheet.py

```diff
@@ -18,7 +19,7 @@
 class Spreadsheet:
     """The class that represents a spreadsheet."""
 
-    def __init__(self, http_client, properties):
+    def __init__(self, http_client: HTTPClient, properties: Dict[str, Any]):
```

Comment:
why is this not caught by `mypy`?

Reply:
I believe this is because we don't use mypy with the `--strict` mode. If a function is not typed then it's ok.

Reply:
I wouldn't recommend it. The strict mode is very strict! It will take us forever to make it pass 🫣

We can do a second pass on the project and type what is missing, may be type de tests too.

After that we could run in strict mode and see what error rises and decide if we want to fix them.

Reply:
then I think for now we can leave out strict mode :)

Reply:
yep, I created this issue to track: adding missing types in code + tests: #1430